### PR TITLE
Feature/rpc layer

### PR DESCRIPTION
**Pull request by tab1tha**

### CODE REVIEW

File: cin_validator/__main__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import click
 import pytest
 
-from cin_validator.cin_validator_class import CinValidationSession
+from cin_validator import cin_validator_class as cin_class
```
Comment (willLPD2I) [inline cin_validator/__main__.py:8]: neat

File: cin_validator/__main__.py [hunk 2]
```diff
@@ -41,7 +41,8 @@ def list_cmd(ruleset):
 @click.option("--output/--no_output", "-o/-no", default=False)
 def run_all(filename: str, ruleset, issue_id, output):
 
-    validator = CinValidationSession(filename, ruleset, issue_id)
+    data_files = cin_class.process_data(filename)
+    validator = cin_class.CinValidationSession(data_files, ruleset, issue_id)
```
Comment (willLPD2I) [inline cin_validator/__main__.py:45]: I think you're right, process date can either go in here, or be part of the process data function itself

File: cin_validator/cin_validator_class.py [hunk 3]
```diff
@@ -8,13 +8,35 @@
...
-    def __init__(self, filename, ruleset, issue_id=None) -> None:
-        # TODO detect filetype xml/csv/zip. check if the directory is a folder.
-        fulltree = ET.parse(filename)
-        root = fulltree.getroot()
-        self.data_files_obj = DataContainerWrapper(XMLtoCSV(root))
+    def __init__(
+        self, data_files=None, ruleset="rules.cin2022_23", issue_id=None
+    ) -> None:
 
+        self.data_files = data_files
```
Comment (willLPD2I) [inline cin_validator/cin_validator_class.py:39]: nice

**Review by willLPD2I — APPROVED** [review]
All checks out okay, what's with the empty file?

**Review by dotloadmovie — APPROVED** [review]

**Pull request merged**